R/plotpercentile.R

```diff
@@ -40,31 +42,46 @@ plotpercentile <- function(x){
     aes <- ggplot2::aes
     element_blank <- ggplot2::element_blank
 
-    p<- ggplot2::ggplot(data=plotdata[plotdata$type=="Permutation", ], aes(x=x, y=y, group=label), aspect="iso") +
-       ggplot2::scale_colour_gradientn("log(p-val)",
-                                       colours=palette$grpalette,
-                                       values = palette$values,
-                                       limits = palette$limits
-       ) +
-       ggplot2::geom_path(color="grey80") +
-       ggplot2::geom_path(data=plotdata[plotdata$type=="Percentile", ], aes(x=x, y=y, group=label), color="grey40", size=1.25)+
-       ggplot2::geom_path(data=plotdata[plotdata$type=="Actual", ], aes(x=x, y=y, group=label, color = cc.10))+
-       ggplot2::geom_point(data=plotdata[plotdata$type=="Actual", ], aes(x=x, y=y, group=label, color = cc.10), size=2) +
-       ggplot2::theme_bw() +
-       ggplot2::theme(axis.title.x = element_blank(),
+    p<- ggplot2::ggplot(data=plotdata[plotdata$type=="Permutation", ], aes(x=x, y=y, group=label), aspect="iso")
+    if (coloring %in% c("p","p.adj","fish")) {
+      p <- p +
+        ggplot2::scale_colour_gradientn("log(p-val)",
+                                        colours=palette$grpalette,
+                                        values = palette$values,
+                                        limits = palette$limits
+       )
+    } else if (coloring != "Terms") {
+      p <- p +
+        ggplot2::scale_colour_gradient(coloring)
+    }
+    p <- p +
+      ggplot2::geom_path(color="grey80") +
+      ggplot2::geom_path(data=plotdata[plotdata$type=="Percentile", ], aes(x=x, y=y, group=label), color="grey40", size=1.25)
+    if (coloring != "Terms") {
+      p <- p +
```

Comment:
This condition and the else differ only in the color that you are passing. There's no point in duplicating the complete set of lines. Just set the color and then pass it as a variable.

Reply:
Not sure which lines you are referring to.  Lines 62-63 set colour as an aesthetic using a variable (color).  Lines 66-67 specify a fixed color ("blue").

Reply:
Can't you set the fixed "blue" color in the variable if unspecified, and invoke the plot lines only once, always with the variable?

Reply:
I don't know how ggplot would interpret color being defined both inside and outside aes(),  I think that this approach is safer and more transparent.